nahusenayElias/Team_4
# DRD-28 Connect Mautic to Drupal

## In this PR:
- Installed and enabled Mautic and Mautic Paragraphs module
- Mautic script added to site footer to collect data from the user

**NOTE! Mautic Paragraph module has not been configured yet. I'll keep working on that and make a separate ticket + PR for it.** 

**NOTE! Try different browsers or use private or incognito mode for testing, Firefox doesn't register page visits in normal mode** 

### Testing instructions: 
- git checkout branch
- lando start
- lando db-import DRD-28.sql.gz
- lando composer install
- lando drush cim
- cd mautic-lando
- lando start
- go to Drupal UI and check that the modules are installed and enabled (Admin > Extend > search for 'mautic' and you should see the two modules installed and enabled)
![image](https://github.com/user-attachments/assets/1425b5d0-514c-4327-a42e-5733c5fa1e7d)
- Check the Mautic module configuration and update the localhost port of your Mautic (Admin > Configuration > System > Mautic)
![Screenshot 2024-10-29 at 10-16-44 Mautic Drush Site-Install](https://github.com/user-attachments/assets/fbbbed5a-5358-4477-bffb-25267e450772)
- Go to Mautic dashboard
- Click on settings
- Configuration
- API settings > API enabled > Yes
- System settings > Cors settings > restrict domains > no
- Tracking settings  >  Identify visitor by tracking url > Yes
- Tracking settings  >  Identify visitors by IP > Yes
- Click around the pages on the Drupal side and you should see the page visits on your Mautic dashboard and in the Contacts tab.
